daml-script/daml/Daml/Script.daml

```diff
@@ -218,6 +220,22 @@ queryContractId p c = lift $ Free $ QueryContractId QueryContractIdPayload with
   continue = pure . fmap (fromSome . fromAnyTemplate)
   locations = getCallStack callStack
 
+data QueryViewPayload a = QueryViewPayload
+  { parties : [Party]
+  , interfaceId : TemplateTypeRep
+  , continue : [LedgerValue] -> a
+  , locations : [(Text, SrcLoc)]
+  } deriving Functor
+
+-- | Query the set of active contracts views for an interface
+-- that are visible to the given party.
+queryView : forall i v p. (TemplateOrInterface i, HasInterfaceView i v, IsParties p) => p -> Script [(ContractId i, v)]
```

Comment:
Note, the computation of a view may fail. You will have to handle this case (in another PR)

To follow, the convention we used in Trigger you will have to change the signature as follow:
```suggestion
queryView : forall i v p. (TemplateOrInterface i, HasInterfaceView i v, IsParties p) => p -> Script [(ContractId i, Optional v)]
```

Reply:
> Note, the computation of a view may fail. You will have to handle this case (in another PR)
Yes. I have a TODO in the code, and a checklist item in issue #14830

I will change the type for `queryView` as you suggest.

Reply:
see #15384